go/arrow/array/extension.go

```diff
@@ -38,7 +38,8 @@ type ExtensionArray interface {
 	ExtensionType() arrow.ExtensionType
 	// Storage returns the underlying storage array for this array.
 	Storage() arrow.Array
-
+	// ValueString returns a string represenation of the value at the given index for the extension array.
+	ValueString(i int) string
```

Comment:
looks like you need to add a default implementation of `ValueString` for `ExtensionArrayBase` in order for `staticcheck` not to complain that `mustEmbedExtensionArrayBase` is unused. 

Essentially, `ExtensionArrayBase` needs to meet the `ExtensionArray` interface for staticcheck linting to be happy.